### PR TITLE
Always compile autoconf tests locally

### PR DESCRIPTION
**Pull request by kito-cheng**
Compile locally if it seems like a autconf test.

#300

### CODE REVIEW

File: client/arg.cpp [hunk 1]
```diff
@@ -55,6 +55,44 @@ inline int str_startswith(const char *head, const char *worm)
     return !strncmp(head, worm, strlen(head));
 }
 
+/**
+ * Return a pointer to the basename of the file (everything after the
+ * last slash.)  If there is no slash, return the whole filename,
+ * which is presumably in the current directory.
+ **/
+static const char *find_basename(const char *sfile)
```
Comment (llunak) [inline client/arg.cpp]: There already is find_basename() function (utils.h).

File: client/arg.cpp [hunk 2]
```diff
@@ -55,6 +55,44 @@ inline int str_startswith(const char *head, const char *worm)
...
+
+    if (slash == NULL || slash[1] == '\0')
+        return sfile;
+
+    return slash + 1;
+}
+
+/* Some files should always be built locally... */
+static bool
+source_needs_local(const string &filename)
```
Comment (llunak) [inline client/arg.cpp]: Source needs local what? Please use a more descriptive name, e.g. should_always_build_locally() or somesuch.

File: client/arg.cpp [hunk 3]
```diff
@@ -55,6 +55,44 @@ inline int str_startswith(const char *head, const char *worm)
...
+{
+    const char *p;
+
+    p = find_basename(filename.c_str());
+
+    if (str_startswith("conftest.", p) || str_startswith("tmp.conftest.", p)) {
+#if CLIENT_DEBUG
```
Comment (llunak) [inline client/arg.cpp]: It is fine to always have this info message, the #if shouldn't be there.

File: client/arg.cpp [hunk 4]
```diff
@@ -55,6 +55,44 @@ inline int str_startswith(const char *head, const char *worm)
...
+        log_info() << "autoconf tests are run locally: " << p << endl;
+#endif
```
Comment (llunak) [inline client/arg.cpp]: The message should be shown normally (if debug is set to the info level), the #if shouldn't be there.
Also, the right place for this message should be where the local build is forced (i.e. the "always_local = true" in the other change), as that is the place where the action takes place. If this function got called from multiple place for whatever reason, it'd print the message multiple times.

**Comment by llunak** [comment]
Generally looks good, but could you please first do the suggested cleanups?

**Comment by kito-cheng** [comment]
Thanks your review, I've updated according your comment :)

**Comment by llunak** [comment]
Ok, thank you for the patch.

**Pull request merged**